File: homework/lesson_07/Operations.java
package homework.lesson_07;

public class Operations extends Calculator{
    public Operations(double num1, double num2, String symbol) {
        super(num1, num2, symbol);
    }

    public String operation(String symbol) {
        switch (symbol) {
            case "+":
                return (String.format("%.1f + %.1f = %.1f", getNum1(), getNum2(), sum()));
            case "-":
                return String.format("%.1f - %.1f = %.1f", getNum1(), getNum2(), subtraction());
            case "*":
                return String.format("%.1f * %.1f = %.1f", getNum1(), getNum2(), multiply());
            case "/":
                return String.format("%.1f / %.1f = %.1f", getNum1(), getNum2(), devide());
            default:
                return null;
        }
    }

}

File: homework/lesson_07/Operations.py
from Calculator import Calculator


class Operation(Calculator):
    def __init__(self, num_1, second_number, symbol):
        self._num_1 = num_1
        self._num_2 = second_number
        self._symbol = symbol

    def operation(self, symbol):
        match symbol:
            case "+":
                return self.sum()
            case "-":
                return self.subtraction()
            case "*":
                return self.multiply()
            case "/":
                return self.devide()
            case _:
                print("Invalid operator")